uni/lib/view/restaurant/widgets/restaurant_page_card.dart

```diff
@@ -43,13 +47,50 @@ class CardFavoriteButton extends StatelessWidget {
       builder: (context, restaurantProvider) {
         final isFavorite =
             restaurantProvider.favoriteRestaurants.contains(restaurant.name);
+        final favoriteCardTypes =
+            Provider.of<HomePageProvider>(context).favoriteCards;
```

Comment:
This is a one-time read of a prop that you need at callback time. Consider moving this to the `onPressed` callback, and use the safer `context.read` option